### PR TITLE
[jit] Avoid passing a vtable argument to DIM methods when making calls out of gsharedvt methods.

### PR DESCRIPTION
**Pull request by vargaz**
Fixes https://github.com/mono/mono/issues/18276.

### CODE REVIEW

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by vargaz** [comment]
@monojenkins build failed

File: mono/mini/method-to-ir.c [hunk 1]
```diff
@@ -7632,16 +7632,15 @@ mono_method_to_ir (MonoCompile *cfg, MonoMethod *method, MonoBasicBlock *start_b
...
 					} else if (mono_class_is_interface (cmethod->klass) && !imt_arg) {
 						/* This can happen when we call a fully instantiated iface method */
 						g_assert (will_have_imt_arg);
 						imt_arg = emit_get_rgctx_method (cfg, context_used,
 														 cmethod, MONO_RGCTX_INFO_METHOD);
 						g_assert (imt_arg);
-						vtable_arg = NULL;
 					}
+					/* This is not needed, as the trampoline code will pass one, and it might be passed in the same reg as the imt arg */
+					vtable_arg = NULL;
```
Comment (lambdageek) [inline mono/mini/method-to-ir.c]: Should we move `g_assert (imt_arg)` from both if statements above to here?

Comment (vargaz) [reply mono/mini/method-to-ir.c]: Its an if else if, so there could be a third case.

Comment (lambdageek) [reply mono/mini/method-to-ir.c]: Ok, so in the third case we can end up with `vtable_arg == NULL` and `imt_arg == NULL`?  That doesn't seem ok.

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by lambdageek** [comment]
@monojenkins backport to 2019-12

**Comment by lambdageek** [comment]
@monojenkins backport to 2019-10

**Pull request merged**